apps/guardian-ui/src/components/GuardianAuthenticationCode.tsx

```diff
@@ -0,0 +1,138 @@
+import React, { useState } from 'react';
+import {
+  Text,
+  useTheme,
+  Modal,
+  ModalOverlay,
+  ModalContent,
+  ModalHeader,
+  ModalBody,
+  ModalCloseButton,
+  Flex,
+  Button,
+} from '@chakra-ui/react';
+import { useTranslation } from '@fedimint/utils';
+import QRCode from 'qrcode.react';
+
+const QR_CODE_SIZE = 256;
+
+interface GuardianAuthenticationCodeProps {
+  federationId: string;
+  ourPeer: { id: number; name: string };
+}
+
+export const GuardianAuthenticationCode: React.FC<
+  GuardianAuthenticationCodeProps
+> = ({ federationId, ourPeer }) => {
+  const theme = useTheme();
+  const { t } = useTranslation();
+  const [isOpen, setIsOpen] = useState(false);
+  const [qrValue, setQrValue] = useState<string>('');
+  const [isAcknowledged, setIsAcknowledged] = useState(false);
+
+  const calculateGuardianAuthenticationCode = () => {
+    const password = sessionStorage.getItem('guardian-ui-key');
+    const params = new URLSearchParams({
+      federationId: federationId,
+      peerId: ourPeer?.id.toString(),
+      guardianName: ourPeer?.name,
+      password: password || '',
```

Comment:
Should we show the QR code without a password if it's actually empty for some reason? Or should we have some kind of failure state? Not sure how likely it actually is so ignore if not worth considering